### PR TITLE
[MM-57106] Remove unused cluster settings

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
The settings `StreamingPort`, `MaxIdleConns`, `MaxIdleConnsPerHost` and `IdleConnTimeoutMilliseconds` were unused and had no effect. This PR cleans them up from the codebase.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-57106

#### Release Note
```release-note
Remove unused cluster settings: `StreamingPort`, `MaxIdleConns`, `MaxIdleConnsPerHost`, `IdleConnTimeoutMillise.conds`
```

### CODE REVIEW

File: server/channels/store/searchlayer/layer_test.go [hunk 1]
```diff
@@ -33,7 +33,7 @@ func TestUpdateConfigRace(t *testing.T) {
 
 	cfg := &model.Config{}
 	cfg.SetDefaults()
-	cfg.ClusterSettings.MaxIdleConns = model.NewInt(1)
+	cfg.ClusterSettings.GossipPort = model.NewInt(9999)
```
Comment (hanzei) [inline server/channels/store/searchlayer/layer_test.go:36]: Needed another setting for the test

**Review by agnivade — APPROVED** [review]
Feel good cleanup!

**Review by streamer45 — APPROVED** [review]
Cool :+1:

**Comment by hanzei** [comment]
E2E tests look good. Merging.

**Comment by hanzei** [comment]
@cwarnermm Heads up that doc updates are required for this PR. Please let me know if you need anything from me. Should be straightforward to move the removed cluster settings to the legacy config page.

**Pull request merged**